cosmos/x/evm/plugins/historical/historical_data.go

```diff
@@ -47,8 +47,8 @@ func (p *plugin) StoreBlock(block *coretypes.Block) error {
 	prefix.NewStore(store, []byte{types.BlockHashKeyToNumPrefix}).Set(block.Hash().Bytes(), numBz)
 
 	// store the version offchain for consistency.
-	if sdk.BigEndianToUint64(store.Get([]byte{types.VersionKey})) != blockNum-1 {
-		panic("off-chain store's latest block number is not synced")
+	if offChainNum := sdk.BigEndianToUint64(store.Get([]byte{types.VersionKey})); offChainNum != blockNum-1 {
```

Comment:
remove, historical plug-in shouldn't have a genesis.

Reply:
you might wanna write version 0 ? tbh i'll think on this.

Reply:
we might wanna make it snapshottable via the handler thing

Reply:
the version key check existed already to panic on height mismatch

Reply:
we can still add those things in future PR